### PR TITLE
Changes after review of multiclass adaboost

### PR DESCRIPTION
**Pull request by orazve**

### CODE REVIEW

**Comment by prutskov** [comment]
make test

**Comment by ShvetsKS** [comment]
make test

**Comment by ShvetsKS** [comment]
make test

File: algorithms/kernel/adaboost/adaboost_train_impl.i [hunk 1]
```diff
@@ -315,10 +314,10 @@ services::Status AdaBoostTrainKernelNew<method, algorithmFPType, cpu>::adaboostS
     DAAL_CHECK(aErrFlag.get(), services::ErrorMemoryAllocationFailed);
 
     /* Initialize weights */
-    for (size_t i = 0; i < nVectors; i++)
+    threader_for(nVectors, nVectors, [&](const size_t i)
```
Comment (SmirnovEgorRu) [inline algorithms/kernel/adaboost/adaboost_train_impl.i]: I think this type of threader_for leads to huge overhead.

**Comment by SmirnovEgorRu** [comment]
Olga, I see many changes in review which were done by other people in the past. So, please, make rebase to eliminate this.

**Comment by orazve** [comment]
@SmirnovEgorRu Thank you, I have made merge from master. Looks like here are only reasonable changes now.

File: algorithms/kernel/adaboost/adaboost_train_impl.i (hunk 1 above)
Comment (orazve) [reply algorithms/kernel/adaboost/adaboost_train_impl.i]: It depends on the number of vectors nad it was the request from Mikhail.

Comment (SmirnovEgorRu) [reply algorithms/kernel/adaboost/adaboost_train_impl.i]: My concern is that you create nVectors tasks.
In this case you should use this function already optimized:
T* service_memset(T * const ptr, const T value, const size_t num)

**Comment by orazve** [comment]
http://intel-ci.intel.com/e9e4e803-01b5-f198-ab63-8cdcd4b723a1
http://intel-ci.intel.com/e9e4df47-4932-f1bb-bc96-8cdcd4b723a1

File: include/algorithms/logistic_regression/logistic_regression_predict_types.h [hunk 2]
```diff
@@ -226,6 +226,9 @@ typedef services::SharedPtr<const Result> ResultConstPtr;
 
 } // namespace interface1
 using interface1::Input;
+using classifier::Parameter;             /* Support of static backward compatibility */
```
Comment (ShvetsKS) [inline include/algorithms/logistic_regression/logistic_regression_predict_types.h]: don`t forget to make related changes in the testing branch and static BC test branches.

**Comment by SmirnovEgorRu** [comment]
make test

**Comment by orazve** [comment]
make test

**Comment by orazve** [comment]
make test

**Pull request merged**